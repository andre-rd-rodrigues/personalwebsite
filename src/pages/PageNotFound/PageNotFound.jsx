import React from "react";
import { Icon } from "@iconify/react";
import PageContainer from "components/PageContainer/PageContainer";
import { useTranslation } from "react-i18next";
import styles from "./pagenotfound.module.scss";

const PageNotFound = () => {
  const { t } = useTranslation();

  return (
    <PageContainer>
      <div className={styles.container}>
        <h1>404</h1>
        <h2>{t("404.title")}</h2>
        <p>{t("404.description")}</p>
        <button
          className={styles.backButton}
          onClick={() => window.history.back()}
        >
          <Icon
            icon="heroicons:arrow-long-left-solid"
            style={{ fontSize: "40px" }}
          />
          {t("404.cta")}
        </button>
      </div>
    </PageContainer>
  );
};

export default PageNotFound;
