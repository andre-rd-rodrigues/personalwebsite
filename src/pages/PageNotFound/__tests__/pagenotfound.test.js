import { render } from "utils/tests/test-utils";
import PageNotFound from "../PageNotFound";

describe("404 page", () => {
  it("renders page correctly", () => {
    const { asFragment } = render(<PageNotFound />);

    expect(asFragment()).toMatchSnapshot();
  });
});
