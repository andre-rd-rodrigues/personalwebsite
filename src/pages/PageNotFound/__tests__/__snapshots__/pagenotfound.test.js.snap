// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 page renders page correctly 1`] = `
<DocumentFragment>
  <div
    class="container undefined"
  >
    <div
      class="wrapper"
      style="opacity: 0;"
    >
      <div
        class="container"
      >
        <h1>
          404
        </h1>
        <h2>
          Page not found
        </h2>
        <p>
          The page you're looking maybe removed or the inserted link doesn't match any existing page.
        </p>
        <button
          class="backButton"
        >
          <span />
          GO BACK
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
